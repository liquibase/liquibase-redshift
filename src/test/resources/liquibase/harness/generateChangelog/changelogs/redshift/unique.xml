<?xml version="1.1" encoding="UTF-8" standalone="no"?>
<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-latest.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
    <changeSet author="test" id="1">
        <createTable tableName="unique_test_table">
            <column name="id" type="INTEGER"/>
            <column name="email" type="VARCHAR(255)"/>
        </createTable>
        <addUniqueConstraint columnNames="email" constraintName="unique_test_table_email_key" tableName="unique_test_table"/>
    </changeSet>
</databaseChangeLog>