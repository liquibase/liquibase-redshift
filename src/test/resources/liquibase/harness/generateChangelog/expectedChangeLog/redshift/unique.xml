<?xml version="1.1" encoding="UTF-8" standalone="no"?>
<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-latest.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
    <changeSet author="yodzh (generated)" id="1674562660276-1">
        <createTable tableName="test_table">
            <column name="id" type="INTEGER"/>
        </createTable>
    </changeSet>
    <changeSet author="yodzh (generated)" id="1674562660276-2">
        <addUniqueConstraint columnNames="id" constraintName="test_unique_constraint" tableName="test_table"/>
    </changeSet>
</databaseChangeLog>
