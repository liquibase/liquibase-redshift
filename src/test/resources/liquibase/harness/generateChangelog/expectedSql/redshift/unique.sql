CREATE TABLE "test_table" ("id" INTEGER);

ALTER TABLE "test_table" ADD CONSTRAINT "test_unique_constraint" UNIQUE ("id");