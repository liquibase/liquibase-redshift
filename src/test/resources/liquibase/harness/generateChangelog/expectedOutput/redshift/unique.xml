<?xml version="1.0" encoding="UTF-8"?>
<databaseChangeLog
        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
    <changeSet id="1" author="as">
        <createTable tableName="test_table">
            <column name="id" type="INTEGER"/>
        </createTable>
        <addUniqueConstraint tableName="test_table" 
                           columnNames="id"
                           constraintName="test_unique_constraint"/>
        <rollback>
            <dropUniqueConstraint tableName="test_table"
                                constraintName="test_unique_constraint"/>
            <dropTable tableName="test_table"/>
        </rollback>
    </changeSet>
</databaseChangeLog>